### ISSUE TITLE
Error getting packages with environment constraints could be more helpful

### ISSUE THREAD

**Issue by briankung** [dart-lang/pub#1864]
A user [on StackOverflow](https://stackoverflow.com/questions/49680136/getting-a-simple-app-to-work-with-cloud-firestore) encountered an issue where a package wasn't available to the Flutter / Dart channel that they were using. The error message in this case is not very specific about what's wrong with the package:

```
Package cloud_firestore has no versions that match >=0.4.0 <0.5.0 derived from:
- flutter_cloud_firestore depends on version ^0.4.0
pub get failed (1)
exit code 1
```

And as Igor noted in their StackOverflow question:

> According to [this](https://pub.dartlang.org/packages/cloud_firestore#-versions-tab-) page, 0.4.0 is the current version. What gives?

The error message might be more helpful if it included versions of flutter or channels like so:

```
Package cloud_firestore has no versions that match >=0.4.0 <0.5.0 in channel beta. Derived from:
- flutter_cloud_firestore depends on version ^0.4.0
pub get failed (1)
exit code 1
```

**Comment by nex3** [comment]
This should be fixed with the new solver, which is available as of Dart 2.0.0-dev.44.0.

**Comment by briankung** [comment]
Oh, I think just read [your post on it!](https://medium.com/@nex3/pubgrub-2fb6470504f) Thank you for the hard work and the write-up!
